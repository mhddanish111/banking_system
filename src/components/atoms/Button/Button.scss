@import '../../../styles/colors.scss';
@import '../../../styles/boarder.scss';
@import '../../../styles/spacing.scss';
@import '../../../styles/mixins.scss';
@import '../../../styles/size.scss';
@import '../../../styles/fonts.scss';
button {
    @include font-body;
    border: none;
    background-color: $color-link;
    border-radius: $border-radius;
    color: $color-font-light;
    height: $height-medium;
    padding: 0 $space-l;
    text-transform: uppercase;
    line-height: $line-height-button;
    &.active {
      outline: none;
    }
    &:hover {
      background-color: $color-link-hover;
      transition: $hover-transition;
      box-shadow: none;
      cursor: pointer;
    }
    &:disabled {
      background-color: $color-disabled;
      cursor: default;
    }
    &.dark {
      background: transparent;
      border: 1px solid $color-white;
      &:hover {
        background-color: $color-white;
        color: $color-font-medium;
        transition: $hover-transition;
      }
    }
  }