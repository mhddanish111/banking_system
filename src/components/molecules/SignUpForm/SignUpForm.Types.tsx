export enum IncomeEnum {
  belowOne = 'below 1 Lac',
  oneToFive = '1 Lac to 5 Lac',
  fiveToTen = '5 Lac to 10 Lac',
  moreThanTen = 'more than 10 Lac',
}

export enum GovtIdEnum {
  aadharCard = 'Aadhar Card',
  panCard = 'Pan Card',
  passport = 'Passport',
  electionIdCard = 'Election Commission ID Card',
}

export type FormData = {
  firstName: string;
  lastName: string;
  email: string;
  mobile: number;
  address: string;
  city: string;
  state: string;
  postalCode: number;
  occupation: string;
  income: IncomeEnum;
  password: string;
  confirmPassword: string;
  govtId: GovtIdEnum;
  govtIdNumber: string;
  aadhaarNumber: string;
  panNumber: string;
};

export type CaptchaProps = {
  valid: boolean;
  showErrorMsg: boolean;
};

export type SignUpFormProps = {
  SignUpHandler: (e: any, data: FormData) => void;
};
