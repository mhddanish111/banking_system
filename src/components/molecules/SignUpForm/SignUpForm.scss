@import "../../../styles/spacing.scss";
@import '../../../styles/boarder.scss';
@import "../../../styles/colors.scss";
.sign-up-form {
  width: 100%;
  display: flex;
  justify-content: center;
  padding: 20px;

  button {
    width: 100%;
  }

  .form-input {
    display: flex;
    flex-direction: column;
    margin-bottom: $space-m;
  }
  .form-action {
    margin-top: $space-m
  }

  .sign-in-link {
    margin-top: $space-xs;
    margin-bottom: $space-m;
  }

  .error-message {
    color: $color-red-50;
  }
}
