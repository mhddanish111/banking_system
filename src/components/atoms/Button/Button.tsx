import React from 'react';
import './Button.scss';

type buttonProps = {
  className?: string;
  children?: React.ReactNode | String;
  type?: 'submit' | 'reset' | 'button';
  ariaLabel?: string;
  disabled?: boolean;
  others?: any;
  onClick?: any;
};

const Button: React.FC<buttonProps> = ({
  className,
  children,
  type,
  disabled,
  ariaLabel,
  ...others
}) => (
  <button
    aria-label={ariaLabel}
    className={className}
    disabled={disabled}
    type={type}
    {...others}
  >
    {children}
  </button>
);

export default Button;
