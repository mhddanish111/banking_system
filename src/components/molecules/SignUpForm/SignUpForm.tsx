import React, { useState } from 'react';
import { isEmpty } from 'lodash';
import { Link } from 'react-router-dom';
import { useForm } from 'react-hook-form';
import Button from '../../atoms/Button';
import Captcha from '../../atoms/Captcha';
import { FormData, CaptchaProps, SignUpFormProps } from './SignUpForm.Types';
import './SignUpForm.scss';

const SignUpForm: React.FC<SignUpFormProps> = ({ SignUpHandler }) => {
  const [password, setPassword] = useState<String>();
  const [govId, setGovId] = useState<String>('');
  const [reCaptchaStatus, setReCaptchaStatus] = useState<CaptchaProps>({
    valid: false,
    showErrorMsg: false,
  });
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<FormData>({ mode: 'onChange' });
  const onSubmit = (data: FormData, e: any) => {
    if (!reCaptchaStatus.valid) {
      return setReCaptchaStatus({ valid: false, showErrorMsg: true });
    }
    if (data.password !== data.confirmPassword) {
      return;
    }
    SignUpHandler(e, data);
  };
  const getErrorMessage = (message: String | undefined) => {
    return message && <p className="error-message">{message}</p>;
  };
  return (
    <form
      name="sign-up-form"
      className="row sign-up-form"
      onSubmit={handleSubmit(onSubmit)}
      autoComplete="off"
      aria-label="sign in form"
    >
      <div className="col-sm-6 col-xs-12">
        <label htmlFor="firstName" aria-label="first name">
          First Name
        </label>
        <input
          type="text"
          id="firstName"
          autoComplete="off"
          {...register('firstName', {
            required: 'First name is required',
            maxLength: {
              value: 50,
              message: 'You exceeded max length of 5 !',
            },
          })}
        />
        {getErrorMessage(errors?.firstName?.message)}
      </div>
      <div className="col-sm-6 col-xs-12">
        <label htmlFor="lastName" aria-label="last name">
          Last Name
        </label>
        <input
          type="text"
          id="lastName"
          autoComplete="off"
          {...register('lastName', {
            required: 'Last name is required',
            maxLength: {
              value: 50,
              message: 'You exceeded max length of 50 !',
            },
          })}
        />
        {getErrorMessage(errors?.lastName?.message)}
      </div>
      <div className="col-sm-6 col-xs-12">
        <label htmlFor="email" aria-label="email">
          Email
        </label>
        <input
          type="email"
          id="email"
          autoComplete="off"
          {...register('email', {
            required: 'Email is required',
            maxLength: {
              value: 50,
              message: 'You exceeded max length of 50 !',
            },
          })}
        />
        {getErrorMessage(errors?.email?.message)}
      </div>

      <div className="col-sm-6 col-xs-12">
        <label htmlFor="mobile" aria-label="mobile number">
          Mobile
        </label>
        <input
          type="number"
          id="mobile"
          autoComplete="off"
          {...register('mobile', {
            required: 'Mobile is required',
            maxLength: {
              value: 10,
              message: 'You exceeded max length of 10 !',
            },
          })}
        />
        {getErrorMessage(errors?.mobile?.message)}
      </div>

      <div className="col-sm-6 col-xs-12">
        <label htmlFor="occupation" aria-label="occupation">
          Occupation
        </label>
        <input
          type="text"
          id="occupation"
          autoComplete="off"
          {...register('occupation', {
            required: 'Occupation is required',
            maxLength: {
              value: 50,
              message: 'You exceeded max length of 50 !',
            },
          })}
        />
        {getErrorMessage(errors?.occupation?.message)}
      </div>

      <div className="col-sm-6 col-xs-12">
        <label htmlFor="income" aria-label="income">
          Income
        </label>
        <select {...register('income', { required: 'Income is required' })}>
          <option value="">Select</option>
          <option value="belowOne">below 1 Lac</option>
          <option value="oneToFive">1 Lac to 5 Lac</option>
          <option value="fiveToTen">5 Lac to 10 Lac</option>
          <option value="moreThanTen">more than 10 Lac</option>
        </select>
        {getErrorMessage(errors?.income?.message)}
      </div>

      <div className="col-sm-6 col-xs-12">
        <label htmlFor="aadhaarNumber" aria-label="aadhaar number">
          Aadhaar Number
        </label>
        <input
          type="text"
          id="aadhaarNumber"
          autoComplete="off"
          {...register('aadhaarNumber', {
            required: 'aadhaar number is required',
            // pattern: {
            //   value: /^[2-9]{1}[0-9]{3}\s[0-9]{4}\s[0-9]{4}$/,
            //   message: "Aadhaar number is not valid",
            // },
            maxLength: {
              value: 15,
              message: 'You exceeded max length!',
            },
          })}
        />
        {getErrorMessage(errors?.aadhaarNumber?.message)}
      </div>

      <div className="col-sm-6 col-xs-12">
        <label htmlFor="panNumber" aria-label="pan card number">
          Pan card Number
        </label>
        <input
          type="text"
          id="panNumber"
          autoComplete="off"
          {...register('panNumber', {
            required: 'pan card number is required',
            // pattern: {
            //   value: /^[A-Z]{5}[0-9]{4}[A-Z]{1}$/,
            //   message: "Pan card number is not valid",
            // },
            maxLength: {
              value: 10,
              message: 'You exceeded max length!',
            },
          })}
        />
        {getErrorMessage(errors?.panNumber?.message)}
      </div>

      <div
        className={`${!isEmpty(govId) ? 'col-sm-6' : 'col-sm-12'} col-xs-12`}
      >
        <label htmlFor="govtId" aria-label="government id">
          Govt ID
        </label>
        <select
          {...register('govtId', {
            onChange: (e) => setGovId(e.target.value),
          })}
        >
          <option value="">Select</option>
          <option value="passport">Passport</option>
          <option value="electionIdCard">Election Commission ID Card</option>
        </select>
        {getErrorMessage(errors?.govtId?.message)}
      </div>

      {!isEmpty(govId) && (
        <div className="col-sm-6 col-xs-12">
          <label htmlFor="govtIdNumber" aria-label="government ID number">
            Government ID Number
          </label>
          <input
            type="text"
            id="govtIdNumber"
            autoComplete="off"
            {...register('govtIdNumber', {
              required: !isEmpty(govId)
                ? 'Government ID Number is required'
                : false,
              maxLength: {
                value: 30,
                message: 'You exceeded max length of 30 !',
              },
            })}
          />
          {getErrorMessage(errors?.govtIdNumber?.message)}
        </div>
      )}

      <div className="col-sm-6 col-xs-12">
        <label htmlFor="address" aria-label="address">
          Address
        </label>
        <input
          type="text"
          id="address"
          autoComplete="off"
          {...register('address', {
            required: 'Address is required',
            maxLength: {
              value: 50,
              message: 'You exceeded max length of 50 !',
            },
          })}
        />
        {getErrorMessage(errors?.address?.message)}
      </div>

      <div className="col-sm-6 col-xs-12">
        <label htmlFor="city" aria-label="city">
          City
        </label>
        <input
          type="text"
          id="city"
          autoComplete="off"
          {...register('city', {
            required: 'City is required',
            maxLength: {
              value: 50,
              message: 'You exceeded max length of 50 !',
            },
          })}
        />
        {getErrorMessage(errors?.city?.message)}
      </div>

      <div className="col-sm-6 col-xs-12">
        <label htmlFor="state" aria-label="state">
          State
        </label>
        <input
          type="text"
          id="state"
          autoComplete="off"
          {...register('state', {
            required: 'State is required',
            maxLength: {
              value: 50,
              message: 'You exceeded max length of 50 !',
            },
          })}
        />
        {getErrorMessage(errors?.state?.message)}
      </div>

      <div className="col-sm-6 col-xs-12">
        <label htmlFor="postalCode" aria-label="postal code">
          Postal Code
        </label>
        <input
          type="number"
          id="postalCode"
          autoComplete="off"
          {...register('postalCode', {
            required: 'Postal code is required',
            maxLength: {
              value: 6,
              message: 'You exceeded max length of 6 !',
            },
          })}
        />
        {getErrorMessage(errors?.postalCode?.message)}
      </div>

      <div className="col-sm-6 col-xs-12">
        <label htmlFor="password" aria-label="password">
          Password
        </label>
        <input
          type="password"
          id="password"
          autoComplete="off"
          {...register('password', {
            onBlur: (e) => setPassword(e.target.value),
            required: 'Password is required',
            maxLength: {
              value: 15,
              message: 'You exceeded max length of 15 !',
            },
          })}
        />
        {getErrorMessage(errors?.password?.message)}
      </div>

      <div className="col-sm-6 col-xs-12">
        <label htmlFor="confirmPassword" aria-label="confirm password">
          Confirm Password
        </label>
        <input
          type="password"
          id="confirmPassword"
          autoComplete="off"
          {...register('confirmPassword', {
            required: 'Confirm password is required',
            validate: (value) =>
              value === password || 'Password does not match',
            maxLength: {
              value: 15,
              message: 'You exceeded max length of 15 !',
            },
          })}
        />
        {getErrorMessage(errors?.confirmPassword?.message)}
      </div>
      <div className="col-xs-12 form-action">
        <Captcha setReCaptchaStatus={setReCaptchaStatus} />
        {reCaptchaStatus.showErrorMsg &&
          !reCaptchaStatus.valid &&
          getErrorMessage('Captcha is not valid')}
      </div>

      <div className="col-xs-12 form-action">
        <Button type="submit">Sign Up</Button>
      </div>
      <div className="col-xs-12 sign-in-link">
        <Link to="/">Sign In</Link>
      </div>
    </form>
  );
};

export default SignUpForm;
