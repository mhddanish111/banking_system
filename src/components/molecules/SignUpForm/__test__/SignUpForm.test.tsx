import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import Button from '../../../atoms/Button';
import SignUpForm from '../SignUpForm';

describe('<SignUpForm />', () => {
  const SignUpHandler = jest.fn();
  const wrapper = shallow(<SignUpForm SignUpHandler={SignUpHandler} />);
  it('should render the SignUpForm Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call form event', () => {
    const onSubmit = jest.fn();
    const handleSubmit = jest.fn();
    const event = { preventDefault: jest.fn() };
    const data = {
      firstName: 'test',
      lastName: 'test',
      email: 'test@test.com',
      mobile: 99999999,
      address: 'test',
      city: 'test',
      state: 'test',
      postalCode: 99999,
      occupation: 'test',
      income: 'test',
      password: 'test@1234',
      confirmPassword: 'test@1234',
      govtId: 1,
      govtIdNumber: 1231233,
      aadhaarNumber: 2312312,
      panNumber: 'abc123',
    };
    wrapper
      .find('.sign-up-form')
      .simulate('submit', handleSubmit(onSubmit(data, event)));
    expect(onSubmit).toHaveBeenCalled();
  });
  it('should call button event', () => {
    const onClick = jest.fn();
    const event = { preventDefault: jest.fn() };
    wrapper.find(Button).simulate('submit', onClick(event));
    expect(onClick).toHaveBeenCalled();
  });
});
