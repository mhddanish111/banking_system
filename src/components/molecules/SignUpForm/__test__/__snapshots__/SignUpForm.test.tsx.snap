// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> should render the SignUpForm Component correctly 1`] = `
<form
  aria-label="sign in form"
  autoComplete="off"
  className="row sign-up-form"
  name="sign-up-form"
  onSubmit={[Function]}
>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="first name"
      htmlFor="firstName"
    >
      First Name
    </label>
    <input
      autoComplete="off"
      id="firstName"
      name="firstName"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="last name"
      htmlFor="lastName"
    >
      Last Name
    </label>
    <input
      autoComplete="off"
      id="lastName"
      name="lastName"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="email"
      htmlFor="email"
    >
      Email
    </label>
    <input
      autoComplete="off"
      id="email"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      type="email"
    />
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="mobile number"
      htmlFor="mobile"
    >
      Mobile
    </label>
    <input
      autoComplete="off"
      id="mobile"
      name="mobile"
      onBlur={[Function]}
      onChange={[Function]}
      type="number"
    />
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="occupation"
      htmlFor="occupation"
    >
      Occupation
    </label>
    <input
      autoComplete="off"
      id="occupation"
      name="occupation"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="income"
      htmlFor="income"
    >
      Income
    </label>
    <select
      name="income"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value=""
      >
        Select
      </option>
      <option
        value="belowOne"
      >
        below 1 Lac
      </option>
      <option
        value="oneToFive"
      >
        1 Lac to 5 Lac
      </option>
      <option
        value="fiveToTen"
      >
        5 Lac to 10 Lac
      </option>
      <option
        value="moreThanTen"
      >
        more than 10 Lac
      </option>
    </select>
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="aadhaar number"
      htmlFor="aadhaarNumber"
    >
      Aadhaar Number
    </label>
    <input
      autoComplete="off"
      id="aadhaarNumber"
      name="aadhaarNumber"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="pan card number"
      htmlFor="panNumber"
    >
      Pan card Number
    </label>
    <input
      autoComplete="off"
      id="panNumber"
      name="panNumber"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-sm-12 col-xs-12"
  >
    <label
      aria-label="government id"
      htmlFor="govtId"
    >
      Govt ID
    </label>
    <select
      name="govtId"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value=""
      >
        Select
      </option>
      <option
        value="passport"
      >
        Passport
      </option>
      <option
        value="electionIdCard"
      >
        Election Commission ID Card
      </option>
    </select>
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="address"
      htmlFor="address"
    >
      Address
    </label>
    <input
      autoComplete="off"
      id="address"
      name="address"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="city"
      htmlFor="city"
    >
      City
    </label>
    <input
      autoComplete="off"
      id="city"
      name="city"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="state"
      htmlFor="state"
    >
      State
    </label>
    <input
      autoComplete="off"
      id="state"
      name="state"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="postal code"
      htmlFor="postalCode"
    >
      Postal Code
    </label>
    <input
      autoComplete="off"
      id="postalCode"
      name="postalCode"
      onBlur={[Function]}
      onChange={[Function]}
      type="number"
    />
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="password"
      htmlFor="password"
    >
      Password
    </label>
    <input
      autoComplete="off"
      id="password"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    className="col-sm-6 col-xs-12"
  >
    <label
      aria-label="confirm password"
      htmlFor="confirmPassword"
    >
      Confirm Password
    </label>
    <input
      autoComplete="off"
      id="confirmPassword"
      name="confirmPassword"
      onBlur={[Function]}
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    className="col-xs-12 form-action"
  >
    <Captcha
      setReCaptchaStatus={[Function]}
    />
  </div>
  <div
    className="col-xs-12 form-action"
  >
    <Button
      type="submit"
    >
      Sign Up
    </Button>
  </div>
  <div
    className="col-xs-12 sign-in-link"
  >
    <Link
      to="/"
    >
      Sign In
    </Link>
  </div>
</form>
`;
